constraint-solver/src/grouped_expression.rs

```diff
@@ -231,6 +231,28 @@ impl<T: RuntimeConstant, V: Ord + Clone + Eq> GroupedExpression<T, V> {
         }
     }
 
+    /// Splits this expression into head and tail, i.e., `self = head + tail`
+    /// head is the first summand
+    pub fn try_split_head_tail(mut self) -> Option<(Self, Self)> {
+        if self.linear_components().len() + self.quadratic_components().len() < 2 {
```

Comment:
If we remove this, we can also split expressions like `4 * x + 7` into `4 * x` and `7` and thus get access to all the variables.